### PR TITLE
Increase Compatibility of GLO Restart Output

### PR DESCRIPTION
**Pull request by bska**
We need to honour a "small rate threshold" when outputting the lift gas max supply rate limit and total gas production limit.  This threshold renders the output discontinuous, but is needed for compatibility.

### CODE REVIEW

**Comment by bska** [comment]
jenkins build this please

File: src/opm/output/eclipse/AggregateGroupData.cpp [hunk 1]
```diff
@@ -961,24 +967,42 @@ void assignGroupProductionTargets(const Opm::Group&        group,
...
-        ::Opm::RestartIO::Helpers::VectorItems::SGroup::Value::NoGLOLimit;
-
-    if (const auto& max_supply = group.max_lift_gas(); max_supply.has_value()) {
-        sGrp[Ix::GLOMaxSupply] = sgprop(M::gas_surface_rate, max_supply.value());
-    }
-
-    if (const auto& max_total = group.max_total_gas(); max_total.has_value()) {
-        sGrp[Ix::GLOMaxRate] = sgprop(M::gas_surface_rate, max_total.value());
-    }
+    sGrp[Ix::GLOMaxSupply] = getGLORate(std::forward<SGProp>(sgprop), group.max_lift_gas());
```
Comment (atgeirr) [inline src/opm/output/eclipse/AggregateGroupData.cpp:1004]: I am maybe a bit overcautious here, but is it not a bit dangerous in general to std::forward the same argument twice? Maybe the receiving function should take the argument by const reference, to be sure sgprop remains unchanged from one call to the next?

**Comment by bska** [comment]
jenkins build this please

**Comment by tskille** [comment]
This PR has been tested on a full field model set up with advanced features including gas lift optimization. 4 different restart steps has been tested and results are excellent. This PR completes Eclipse compatible restart for a major full field model. @bska, thanks a loot for you patient and stamina on this task.

**Comment by bska** [comment]
> 4 different restart steps has been tested and results are excellent. This PR completes Eclipse compatible restart for a major full field model.

Thanks a lot for the testing, @tskille.  This is really much appreciated.

**Comment by blattms** [comment]
Looks good, now. Merging.

**Pull request merged**